File: app/src/main/java/com/chooloo/www/callmanager/ui/dialpad/DialpadMvpView.java
package com.chooloo.www.callmanager.ui.dialpad;

import com.chooloo.www.callmanager.ui.base.MvpView;

public interface DialpadMvpView extends MvpView {
    boolean isDialer();

    void setNumber(String number);

    void setIsDialer(boolean isCanBeEdited);

    void updateViewModel(String number);

    void registerKeyEvent(int keyCode);

    void backspace();

    void call();

    void callVoicemail();

    void toggleToneGenerator(boolean toggle);

    void stopTone();

    void playTone(int keyCode);

    void toggleCursor(boolean isShow);

    void showDeleteButton(boolean isShow);

    void vibrate();
}

File: app/src/main/java/com/chooloo/www/callmanager/ui/dialpad/DialpadMvpView.kt
package com.chooloo.www.callmanager.ui.dialpad

import com.chooloo.www.callmanager.ui.base.MvpView

interface DialpadMvpView : MvpView {
    val isDialer: Boolean
    var number: String

    fun showDeleteButton(isShow: Boolean)
    fun showAddContactButton(isShow: Boolean)
    fun toggleDialerView(isToggle: Boolean)
    fun registerKeyEvent(keyCode: Int)
    fun setViewModelNumber(number:String?)
    fun backspace()
    fun call()
    fun callVoicemail()
    fun addContact()
    fun vibrate()
    fun playTone(keyCode: Int)
}